languages.adoc

```diff
@@ -18,88 +18,102 @@ SPDX-FileType: DOCUMENTATION
 
-.SDK Modules
-[#content,width="100%",cols="20%,20%,60%",options="header"]
-|===
-| Folder | Specification |Description
-
-| `uri`
-| link:basics/uri.adoc[UUri]
-| How things are addressed in uProtocol
-
-| `uuid`
-| link:basics/uuid.adoc[UUID]
-| Unique ID & timestamp specification 
-
-| `cloudevents`
-| link:up-l1/cloudevents.adoc[CloudEvents]
-| Specification of how uProtocol data models can be expressed using CloudEvents data model
-
-| `transport`
-| link:up-l1/README.adoc[uTransport]
-a| uP-L1 Transport Interface & data model.
-
-* *MUST* declare the interface & data model where as link:../upclient.adoc[uProtocol Client Library] implement the said interface.
-
-| `rpc`
-| link:up-l2/rpcclient.adoc[uP-L2 RPC Client]
-a|APIs used by clients to invoke methods to simplify implementation of the RPC design pattern. 
+[oft-sid="req~up-language-test~1"]
+--
+ * *MUST* be able to run the test suite such that merge requests (commits) cannot be merged till tests are passed
+--
 
-|===
+[oft-sid="req~up-language-ci~1"]
+--
+ * *MUST* have workflows that ensures code compiles before the can be merged
+--
 
+[oft-sid="req~up-language-ci~2"]
+--
+ * *SHOULD* cover 100% of all code and specifications defined in <<Source>>
+--
 
-== Test
+[oft-sid="req~up-language-ci~3"]
+--
+ * *SHOULD* have lint & documentation coverage checks
+--
 
- * *SHOULD* include a test suite under the folder `test`
- * *MUST* be able to run the test suite such that merge requests (commits) cannot be merged till tests are passed
-  * *SHOULD* cover 100% of all code and specifications defined in <<Source>>
 
+=== Build
 
-== Build
+[oft-sid="req~up-language-build~1"]
+--
+ * *MUST* be easily built in both a Windows and Linux development environment.
+--
 
- * *MUST* be easily built in both a Windows and Linux development environment. Any build requirements (i.e. package dependencies) need to be properly documented in the README
-* *MUST* be able to be imported to any wellknown IDE (Eclipse, IntelliJ, VSCode, etc..) and include self contained build system to build the code and dependencies(ex. maven, gradle, clang, soong, etc...).
-* *MUST* fetch external dependency libraries as handle build dependencies, *MUST NOT* copy external dependencies
-* *MUST NOT* depend on external copyright or copyleft libraries/modules, only permissive licenses such as Apache 2.0, MIT, shall be accepted etc..
+[oft-sid="req~up-language-build~2"]
+--
+ * *MUST* include self contained build system to build the code and dependencies(ex. maven, gradle, clang, soong, etc...).
+--
 
+[oft-sid="req~up-language-build~3"]
+--
+ * *MUST* fetch and build external dependency libraries, *MUST NOT* copy external dependencies
+--
 
-== Documentation
+== Class Diagram
 
- * *MUST* include a main README.adoc file that explains how to build and use the library. 
- * Each module *MUST* also have a README.adoc that explains how to use said module
+.Class Diagram
+[#up-libraries]
+image:up_libraries.drawio.svg[width=800]
```

Comment:
in the diagram, all L2 (blue) boxes are defined as interfaces, and have no implementation